contracts/schemes/FundingRequest.sol

```diff
@@ -0,0 +1,158 @@
+pragma solidity ^0.5.16;
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "../libs/StringUtil.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+
+/**
+ * @title A scheme for requesting funding from an organization
+ */
+
+contract FundingRequest is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable {
+    using SafeMath for uint;
+    using StringUtil for string;
+
+    event NewFundingProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address _beneficiary,
+        uint256 _amount,
+        string _descriptionHash
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, bool _decision);
+
+    event Redeem(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount
+    );
+
+    struct Proposal {
+        address beneficiary;
+        uint256 amount;
+        string descriptionHash;
+        uint256 executionTime;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+
+    string public constant FUNDED_BEFORE_DEADLINE_KEY = "FUNDED_BEFORE_DEADLINE";
```

Comment:
having that in 2 places is not so good
does import these values from the JoinAndQuit scheme might work ?
I think it is ..
or lets create an interface file which both scheme can import and use .

Reply:
Updated